client-misk/src/test/kotlin/app/cash/backfila/client/misk/hibernate/SingleInstanceHibernateBackfillTest.kt

```diff
@@ -67,98 +49,105 @@ class UnshardedHibernateBackfillTest {
   @Test fun emptyTable() {
     val run = backfila.createDryRun<TestBackfill>()
         .apply { configureForTest() }
-    assertEquals(run.prepareBackfillResponse.instances.size, 1)
-    assertNull(run.instanceProgressSnapshot.values.single().keyRange.start)
-    // Trying to get a batch on an empty tablet gets nothing to execute.
-    assertTrue { run.singleScan().batches.isEmpty() }
-    assertTrue { run.complete() }
+    assertThat(run).hasSingleInstanceCursorThat().hasKeyRangeStartThat().isNull()
+    assertThat(run).hasSingleInstanceCursorThat().hasKeyRangeEndThat().isNull()
+    // Trying to scan for a batch on an empty tablet gets nothing to execute.
+    assertThat(run.singleScan().batches).isEmpty() // Check the returned scan
+    assertThat(run).isFinishedScanning()
```

Comment:
cool

Reply:
Glad you like it. Not sure I got everything 100% right but I think it's a step in the right direction.

I figure @swankjesse will either love it or hate it. I'm awaiting judgement. ⚖️